### PR TITLE
Initial <model> evolution document.

### PR DESCRIPTION
**Pull request by weinig**
Adds an historical account of `<model>` and proposes possible evolutions.

### CODE REVIEW

**Review by hober — CHANGES_REQUESTED** [review]
This looks great; thanks Sam!

Could you also add a link from the explainer to this document?

File: model/HistoryAndEvolution.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+of the `<model>` as soon as the page loaded. No need to render out a static 
+image of the model (which might use a slightly different renderer than the one
+on the device, causing discontinuity) and no need to go to a completely
+different experience just to interact (e.g. spin and scale) the model. Better
+yet, when transitioning to an AR experience (which we still think makes sense
+fullscreen) the same asset can be used for a seamless transition.
+
+By utilizing the browser itself for this AR experience, we also maintain the
+ability to have websites include this functionality without having to prompt the
+user for access to privacy gated feature like the camera and motion data, which
```
Comment (tomayac) [inline model/HistoryAndEvolution.md:41]:
```suggestion
user for access to privacy gated features like the camera and motion data, which
```

File: model/HistoryAndEvolution.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+otherwise a website would have to request to implement an AR experience
+themselves, and which many sites would not want to gate this functionality on.
+
+With this basic idea in place, we thought through how to achieve something like
+this and we found ourselves quickly thinking about models as just another type
+of media for the browser to be able to show. So we took inspiration from the
+existing `<img>`, `<picture>` and `<video>` elements to try to build an element
```
Comment (tomayac) [inline model/HistoryAndEvolution.md:48]:
```suggestion
existing `<img>`, `<picture>`, and `<video>` elements to try to build an element
```

File: model/HistoryAndEvolution.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+`<model>` being shown, but that doesn’t mean it should remain that way. 
+
+One next step would be to define an object model (or scene graph) that all the
+supported model asset types (USD, glTF, etc.) could be represented by. This
+would likely take the form of new HTML elements to represent the various aspects
+of the model, and perhaps utilizing CSS to control aspects of the style. This
+would allow a completely dynamic 3D scene to be constructed from HTML and for
+direct manipulation of the components by JS for dynamic interactions and
+animations beyond what is encodable into the model formats themselves. The
+obvious parallel is <svg>, which can be loaded as whole document, or built and
```
Comment (tomayac) [inline model/HistoryAndEvolution.md:70]:
```suggestion
obvious parallel is `<svg>`, which can be loaded as whole document, or built and
```

File: model/HistoryAndEvolution.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+inspected element by element (though I am not sure documents are a necessity
+for this).
+
+With this in place, `<model>` would be re-defined as an element that loads and
+builds a scene graph, perhaps sticking it in the shadow DOM of the `<model>`
+element for use by the page. This would mean defining explicit mappings from
+supported formats (USD, glTF) to this scene graph for browser compatibility.
+
+With access to 3D construction pieces now (and not just a replaced element in
+the 2D page as we have with `<model>`, we will have to define how volumes and
```
Comment (tomayac) [inline model/HistoryAndEvolution.md:80]:
```suggestion
the 2D page as we have with `<model>`), we will have to define how volumes and
```

File: model/HistoryAndEvolution.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+cameras are specified, and how they interact with features like 3D CSS
+transforms, which are non-obvious due to 3D CSS transforms’ flattening
+behaviors. 
+
+This will be a big undertaking.
+
+### Scene Graph & Existing 3D Libraries on the Web
+
+There is a rich legacy of amazing libraries for creating and rendering 3D scene
+graphs on the web (three.js](https://threejs.org/), [Babylon JS](https://babylonjs.com/),
```
Comment (tomayac) [inline model/HistoryAndEvolution.md:90]:
```suggestion
graphs on the web [three.js](https://threejs.org/), [Babylon JS](https://babylonjs.com/), 
```

File: model/HistoryAndEvolution.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+### Scene Graph & Existing 3D Libraries on the Web
+
+There is a rich legacy of amazing libraries for creating and rendering 3D scene
+graphs on the web (three.js](https://threejs.org/), [Babylon JS](https://babylonjs.com/), 
+the list goes on and on) using WebGL/WebGPU and one idea would be to try and
+work with the authors of these libraries to see if we can come up with an
+underlying scene graph that we can share between the browser engine and the
+libraries. The goal would be to allow users of these libraries to continue using
+the APIs and tools these libraries have built to build 3D scenes, but then have
+a way to specify that they want to use the browser’s builtin renderer when
```
Comment (tomayac) [inline model/HistoryAndEvolution.md:96]:
```suggestion
a way to specify that they want to use the browser’s built-in renderer when
```

File: model/HistoryAndEvolution.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+appropriate. 
+
+Importantly, it won’t always be the right choice for a website to want to use
+the browsers builtin renderer, as it will come with limitations not present in
```
Comment (tomayac) [inline model/HistoryAndEvolution.md:100]:
```suggestion
the browsers built-in renderer, as it will come with limitations not present in
```

**Review by tomayac — COMMENTED** [review]
Just a drive-by review. Very interesting to see the perspective and the potential horizon for `model`.

**Comment by weinig** [comment]
> This looks great; thanks Sam!
> 
> Could you also add a link from the explainer to this document?

Will do.

**Comment by weinig** [comment]
> Just a drive-by review. Very interesting to see the perspective and the potential horizon for `model`.

Thanks for the edits.

**Comment by weinig** [comment]
Please let me know if the new version works.

**Review by hober — APPROVED** [review]
Looks great!

File: model/HistoryAndEvolution.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+## A short history
+
+The [`<model>` element](https://github.com/WebKit/explainers/tree/main/model)
+was  born out of a desire to take the next step and improve the experience of
+Safari’s integration with iOS’s [AR Quick Look](https://developer.apple.com/augmented-reality/quick-look/)
+feature. AR Quick Look’s existing integration in Safari allows web pages to link
+to .usdz files that when loaded, open up a full screen experience that allows
+the .usdz model to be viewed with limited interaction (translate, rotate and
+scale) either in a virtual environment (a white void) or in AR using the devices
```
Comment (smfr) [inline model/HistoryAndEvolution.md:23]: device's

File: model/HistoryAndEvolution.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+the list goes on and on) using WebGL/WebGPU and one idea would be to try and
+work with the authors of these libraries to see if we can come up with an
+underlying scene graph that we can share between the browser engine and the
+libraries. The goal would be to allow users of these libraries to continue using
+the APIs and tools these libraries have built to build 3D scenes, but then have
+a way to specify that they want to use the browser’s built-in renderer when
+appropriate. 
+
+Importantly, it won’t always be the right choice for a website to want to use
+the browsers built-in renderer, as it will come with limitations not present in
```
Comment (smfr) [inline model/HistoryAndEvolution.md:100]: browser's

**Pull request merged**